core/threaded/reactor_threaded.c

```diff
@@ -234,11 +237,19 @@ bool wait_until(instant_t logical_time, lf_cond_t* condition) {
   }
 #endif
   if (!fast) {
-    // Check whether we actually need to wait, or if we have already passed the timepoint.
-    interval_t wait_duration = wait_until_time - lf_time_physical();
-    if (wait_duration < MIN_SLEEP_DURATION) {
-      LF_PRINT_DEBUG("Wait time " PRINTF_TIME " is less than MIN_SLEEP_DURATION " PRINTF_TIME ". Skipping wait.",
-                     wait_duration, MIN_SLEEP_DURATION);
+    // Subtract the control value from the requested wait_until_time
+    interval_t wait_until_time_with_adjustment = wait_until_time - error_control;
+    instant_t now = lf_time_physical();
+
+    // If the requested time is already passed, return
+    // If the adjusted wait time is already passed, but the requested wait time
+    // didn't come yet, then adjust the control value and busy wait until the requested time.
+    if (wait_until_time < now) {
+      return true;
+    } else if (wait_until_time_with_adjustment < now && wait_until_time > now) {
+      error_control = error_control - (((now - wait_until_time_with_adjustment) * KI_MUL) / KI_DIV);
+      while (wait_until_time > lf_time_physical())
+        ;
       return true;
     }
```

Comment:
@soyerefsane The case `wait_until_time == now` is not covered unless the `else if` branch is entered first.